### ISSUE TITLE
Supabase studio don't allow uploads after change of the digitalocean token & sapces key.

### ISSUE THREAD

**Issue by anasfik** [digitalocean/supabase-on-do#24]
Hello, I was having a fully working supabase instance in my digital ocean droplet, However I did faced a previous issue that I didn't know where I should point, so I thought that I should update/edit the digitalocean token & spaces key, I deleted previous ones from dashboard, then created the new ones, then opened the console UI of my droplet.

`digitalocean.ini` file where i did updated the digitalocean token:
![image](https://user-images.githubusercontent.com/25140579/226147550-e7993070-64b9-4ee3-a84e-3fcd12b4466f.png)

`.env` file where I did updated the spaces keys:
![image](https://user-images.githubusercontent.com/25140579/226147629-f4f3712d-620a-4f60-9c89-a1dc8cb02095.png)

 
I'am sure the keys which are there does exists in the dashboard of digitalocean!

After that I runned `docker-compose.yml up -d` to apply updates.

and other supabase studio aspects such database and auth still works fine, but in the storage when I do try to upload a file it shows me :
![image](https://user-images.githubusercontent.com/25140579/226147786-3f2c685d-9a1a-4b93-a82c-1f07cdc52c0e.png)

it shows me "accessDenied" but I don't seem to specify what is wrong exactly and what I did wrong or is it even related the leys or other thing, please help me, supabase resources are very limited.

**Comment by anasfik** [comment]
Update 2:
when I check the logs of the storage-api container, I get this log about this log about the request:
```json
{
   "level":"info",
   "time":"2023-03-19T11:08:29.057Z",
   "pid":18,
   "hostname":"b08d1e21d305",
   "reqId":"req-4pl",
   "tenantId":"stub",
   "project":"stub",
   "req":{
      "method":"POST",
      "url":"/object/audibles/my_file.mp3",
      "headers":{
         "host":"storage:5000",
         "x_forwarded_proto":"http",
         "x_real_ip":"172.18.0.6",
         "content_length":"2148787",
         "x_client_info":"supabase-js/2.4.1",
         "user_agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36",
         "content_type":"multipart/form-data; boundary=----WebKitFormBoundaryMEtP6MFMq7j2edmB",
         "accept":"*/*",
         "referer":"https://supabase.example.com/project/default/storage/buckets/audibles"
      },
      "hostname":"storage:5000",
      "remoteAddress":"172.18.0.4",
      "remotePort":56762
   },
   "res":{
      "statusCode":400
   },
   "responseTime":137.87750700116158,
   "error":{
      "name":"UnknownError",
      "httpStatusCode":403,
      "originalError":{
         "name":"AccessDenied",
         "$fault":"client",
         "$metadata":{
            "httpStatusCode":403,
            "attempts":1,
            "totalRetryDelay":0
         },
         "Code":"AccessDenied",
         "BucketName":"supabase-e27eaa78a1bd838c",
         "RequestId":"tx000000000000008e8f48b-006416ed2d-3852c4b5-ams3c",
         "HostId":"3852c4b5-ams3c-ams3-zg03",
         "message":"UnknownError"
      }
   },
   "msg":"POST | 400 | 172.18.0.4 | req-4pl | /object/audibles/combined_phrases_week_298_b95f.mp3 | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
}
```

please, any help or guides throgh this ?

**Comment by timothymamo** [comment]
Hey @anasfik,

The `docker compose` command is incorrect, but unsure if that's just a typo.

Also I don't believe that a `docker compose up -d` is enough for the containers to apply the new configuration. You will have to stop and restart the containers or do a `docker compose down` followed by a `docker compose up -d`.

Hope this gets things working.

**Comment by anasfik** [comment]
hello, thank you for your response!

I was able to point out the issue and fix it, I found out that it was my fault that I did it myself.

the `docker-compose up -d` did actually apply the configuration that I set for spaces & access keys of DigitalIcean, but what I was not aware of is the name/id of the spaces bucket that I was using, so due to the insufficient error information ("AccessDenied"), I thought that the issue was with keys and applying changes.

basically, since I was testing deployments, I deployed/deleted many Supabase instances, I created/deleted many buckets with different names/ids, and so of my lack of experience with `docker`, and `docker-compose`. I thought that changing it from DigitalOcean will change it also automatically in the Supabase instance and that the keys are all that should be changed, but **no**, every time I change the spaces bucket, I have to set its name in `SPACES_GLOBAL_S3_BUCKET`, save file changes, and then run `docker-compose up -d` and the issue resolved.

I was searching and tried for days in all available resources before I get to fix it since the error message is really unhelpful and doesn't specify what the issue is (which I really hope I don't face an error like this again), running `docker logs` to check all logs of containers was not helpful also since it was just showing that a request has 403 status code error without any information.
